test/Mono.Linker.Tests/TestCasesRunner/ResultChecker.cs

```diff
@@ -603,6 +603,12 @@ void VerifyLoggedMessages (AssemblyDefinition original, LinkerTestLogger logger)
 							$"Expected to find logged message matching `{expectedMessagePattern}`, but no such message was found.{Environment.NewLine}Logged messages:{Environment.NewLine}{allMessages}");
 					}
 
+					if (attr.AttributeType.Resolve ().Name == nameof (LogContainsExactAttribute)) {
+						var expectedMessage = (string)attr.ConstructorArguments [0].Value;
+						Assert.That (logger.Messages.Where (s => s.Message.Contains (expectedMessage)), Is.Not.Null,
```

Comment:
Isn't this always true? `.Where` will always return a non-null enumerable - so the condition doesn't actually validate anything (If I read it right) - The `.Any` used above is better - or if there's an `Is` predicate for a non-empty collection, use that.